tests/unit/test_collector.py

```diff
@@ -181,24 +194,24 @@ def test_get_machine_type(self, collector_daemon, mac_address, expect_machine_ty
         assert machine_type.value == expect_machine_type
 
     @pytest.mark.parametrize(
-        "machine, hostname",
+        "machine, machine_id",
         [
             ({"hostname": None, "instance-id": "testid"}, "testid"),
             ({"hostname": "None", "instance-id": "testid"}, "testid"),
             ({"hostname": "testhost", "instance-id": "testid"}, "testhost"),
-            ({"hostname": None, "instance-id": None}, None),
-            ({"hostname": None, "instance-id": "None"}, None),
-            ({"hostname": "None", "instance-id": None}, None),
-            ({"hostname": "None", "instance-id": "None"}, None),
+            ({"hostname": None, "instance-id": None}, "None"),
+            ({"hostname": None, "instance-id": "None"}, "None"),
+            ({"hostname": "None", "instance-id": None}, "None"),
+            ({"hostname": "None", "instance-id": "None"}, "None"),
             ({"hostname": "testhost"}, "testhost"),
-            ({"hostname": None}, None),
-            ({"hostname": "None"}, None),
+            ({"hostname": None}, "None"),
+            ({"hostname": "None"}, "None"),
             ({"instance-id": "testid"}, "testid"),
-            ({"instance-id": None}, None),
-            ({"instance-id": "None"}, None),
-            ({}, None)
-        ]
+            ({"instance-id": None}, "None"),
+            ({"instance-id": "None"}, "None"),
+            ({}, "None"),
+        ],
```

Comment:
since we now have `"pending"` as another "forbidden" machine_id, it should be added to this test.